### PR TITLE
Add Field *Filters* into Struct groups.ListOpts

### PR DESCRIPTION
**Pull request by zhangzujian**
For #978

### CODE REVIEW

**Comment by zhangzujian** [comment]
recheck

**Comment by zhangzujian** [comment]
Ready for review.

File: acceptance/openstack/identity/v3/groups_test.go [hunk 1]
```diff
@@ -62,6 +62,62 @@ func TestGroupCRUD(t *testing.T) {
 		tools.PrintResource(t, g.Extra)
 	}
 
+	var found bool
+	for _, group := range allGroups {
+		tools.PrintResource(t, group)
+		tools.PrintResource(t, group.Extra)
+
+		if group.Name == "testgroup" {
```
Comment (jtopjian) [inline acceptance/openstack/identity/v3/groups_test.go]: The test is failing here. The name is randomly generated, such as `ACPTTESTkl7t28j7`. My suggestion would be to  do

```go
if group.Name == newGroup.Name {
```

File: acceptance/openstack/identity/v3/groups_test.go [hunk 2]
```diff
@@ -62,6 +62,62 @@ func TestGroupCRUD(t *testing.T) {
...
+
+	allGroups, err = groups.ExtractGroups(allPages)
+	th.AssertNoErr(t, err)
+
+	found = false
+	for _, group := range allGroups {
+		tools.PrintResource(t, group)
+		tools.PrintResource(t, group.Extra)
+
+		if group.Name == "testgroup" {
```
Comment (jtopjian) [inline acceptance/openstack/identity/v3/groups_test.go]: same here

File: acceptance/openstack/identity/v3/groups_test.go [hunk 3]
```diff
@@ -62,6 +62,62 @@ func TestGroupCRUD(t *testing.T) {
...
+			found = true
+		}
+	}
+
+	th.AssertEquals(t, found, true)
+
+	listOpts.Filters = map[string]string{
+		"name__contains": "stg",
```
Comment (jtopjian) [inline acceptance/openstack/identity/v3/groups_test.go]: This should be:

```go
"name__contains": "TEST",
```

**Comment by zhangzujian** [comment]
Ready for review.

**Pull request merged**